File: filtro.java
document.addEventListener("DOMContentLoaded", function () {
  const checkboxes = document.querySelectorAll(".filtro-item");
  const blocos = document.querySelectorAll(".image-link");

  checkboxes.forEach(checkbox => {
    checkbox.addEventListener("change", () => {
      const filtrosAtivos = Array.from(checkboxes)
        .filter(cb => cb.checked)
        .map(cb => cb.getAttribute("data-classe"));

      blocos.forEach(bloco => {
        const visivel = filtrosAtivos.some(filtro =>
          bloco.classList.contains(filtro)
        );
        bloco.style.display = visivel ? "block" : "none";
      });
    });
  });
});

File: filtro.js
document.querySelectorAll('.filtro-item').forEach(checkbox => {
  checkbox.addEventListener('change', () => {
    const ativos = Array.from(document.querySelectorAll('.filtro-item'))
      .filter(cb => cb.checked)
      .map(cb => cb.dataset.classe);

    document.querySelectorAll('.image-link').forEach(item => {
      const itemClasse = Array.from(item.classList).find(c => ativos.includes(c));
      item.style.display = itemClasse ? '' : 'none';
    });
  });
});